modules/renter/contractor/update.go

```diff
@@ -20,30 +66,12 @@ func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
 		}
 	}
 
-	// archive expired contracts
-	var expired []types.FileContractID
-	for id, contract := range c.contracts {
-		if c.blockHeight > contract.EndHeight() {
-			// No need to wait for extra confirmations - any processes which
-			// depend on this contract should have taken care of any issues
-			// already.
-			expired = append(expired, id)
-			// move to oldContracts
-			c.oldContracts[id] = contract
-		}
-	}
-	// delete expired contracts (can't delete while iterating)
-	for _, id := range expired {
-		delete(c.contracts, id)
-		c.log.Println("INFO: archived expired contract", id)
-	}
-
 	// If we have entered the next period, update currentPeriod
 	// NOTE: "period" refers to the duration of contracts, whereas "cycle"
 	// refers to how frequently the period metrics are reset.
 	// TODO: How to make this more explicit.
 	cycleLen := c.allowance.Period - c.allowance.RenewWindow
-	if c.blockHeight > c.currentPeriod+cycleLen {
+	if c.blockHeight >= c.currentPeriod+cycleLen {
```

Comment:
This whole block lends itself to a method.  c.invalidateMetricsContract() or similar